recipes/default.rb

```diff
@@ -15,3 +15,54 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+# include_recipe php
+# include_recipe mysql
+
+# Following this tutorial: https://linuxize.com/post/set-up-an-email-server-with-postfixadmin/
+
+group 'vmail' do
+  gid 5000
+end  
+
+user 'vmail' do
+  comment 'Owner of all mailboxes. Used to access emails on this server'
+  home    '/var/mail/vmail'
+  uid     5000
+  gid     5000
+  shell   '/usr/sbin/nologin'
+end
+
+# For testing
+
+user 'www-data' do
+  comment 'Php user'
+  uid     5001
+  gid     5000
+end
+
+directory '/var/www'
+
+remote_file '/var/www/postfixadmin.tar.gz' do
+  source "https://downloads.sourceforge.net/project/postfixadmin/postfixadmin/postfixadmin-#{node['postfixadmin']['version']}/postfixadmin-#{node['postfixadmin']['version']}.tar.gz"
+  mode '0755'
+end
+
+archive_file '/var/www/postfixadmin.tar.gz' do
+  destination '/var/www/tmp'
+  owner 'www-data'
+  not_if { ::File.exist?('/var/www/postfixadmin') }
+end
```

Comment:
Something like the following might be better:

```suggestion
remote_file '/var/www/postfixadmin.tar.gz' do
  source "https://downloads.sourceforge.net/project/postfixadmin/postfixadmin/postfixadmin-#{node['postfixadmin']['version']}/postfixadmin-#{node['postfixadmin']['version']}.tar.gz"
  notifies :extract, 'archive_file '/var/www/postfixadmin.tar.gz', :immediately
end

archive_file '/var/www/postfixadmin.tar.gz' do
  destination '/var/www/postfixadmin'
  strip_components 1
  owner 'www-data'
  action :nothing
end
```
Also,

- Use `Chef::Config[:file_cache_path]` instead of `/var/www` for where you store the tarball. That will put it where we normally put stuff like that
- Make sure you include the checksum

Reply:
I believe strip_components is [new in chef 17.5](https://docs.chef.io/resources/archive_file/)

Reply:
We are using 17.x now, so that's not a problem.

Reply:
How would I test this on `blue`? Kitchen is using cinc client 16 and fails on the strip_components line.